server/controllers/googleApiController.js

```diff
@@ -1,27 +1,97 @@
 const { google } = require("googleapis");
+const Moment = require("moment");
+const MomentRange = require("moment-range");
+const moment = MomentRange.extendMoment(Moment);
 
+// get google authentication
 const oAuth2Client = new google.auth.OAuth2(
-  "871373961261-rjej65g97dc3o6jiuflq6s2gp5v9ptut.apps.googleusercontent.com",
-  "XauLtqJv2LbkiRd7nTMhRnQq",
-  "http://localhost:3000"
+  process.env.client_id,
+  process.env.client_secret,
+  process.env.redirect_uris
 );
+// load auth for getting tokens from google authentication code
 const oauth2 = google.oauth2("v2");
+// load calendar for getting events from google calendar
+const calendar = google.calendar({ version: "v3", auth: oAuth2Client });
 
 module.exports = {
   authenticateUser: function (req, res) {
     oAuth2Client
       .getToken(req.query.code)
       .then(({ tokens }) => {
-        oAuth2Client.setCredentials({ token: tokens });
+        oAuth2Client.setCredentials(tokens);
         return oauth2.userinfo.get({ access_token: tokens.access_token });
       })
       .then((userInfo) => {
         oAuth2Client.setCredentials({
-          token: oAuth2Client.credentials.token,
-          userInfo: userInfo.data,
+          ...oAuth2Client.credentials,
+          ...userInfo.data,
         });
-        res.status(200).json(oAuth2Client.credentials);
+        //TO DO: create user or update user with the user profile and new tokens
+        return res.status(200).json(oAuth2Client.credentials);
       })
       .catch((err) => res.status(422).json(err));
   },
+
+  getAvailability: function (req, res) {
+    const {query} = req;
```

Comment:
For future improvement, you can also expect a query called `timezone` to get passed in from the frontend.

When that is implemented, i think you can plug it into the momentjs date codes and google API call and it will automatically adjust the available time slots 

but we can worry about this after it's more integrated!